client/app/components/FavoritesControl.jsx

```diff
@@ -0,0 +1,89 @@
+import { isFunction } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+import { $rootScope } from '@/services/ng';
+
+function toggleItem(event, item, callback) {
+  event.preventDefault();
+  event.stopPropagation();
+
+  if (item) {
+    if (item.is_favorite) {
+      item.$unfavorite().then(() => {
+        item.is_favorite = false;
+        $rootScope.$broadcast('reloadFavorites');
+        if (isFunction(callback)) {
+          callback();
+        }
+      });
+    } else {
+      item.$favorite().then(() => {
+        item.is_favorite = true;
+        $rootScope.$broadcast('reloadFavorites');
+        if (isFunction(callback)) {
+          callback();
+        }
+      });
+    }
+  }
```

Comment:
This can be simplified:

```javascript
const action = item.is_favorite ? item.$unfavorite : item.$favorite;
action().then(() => {
  item.is_favorite = !item.is_favorite;
  $rootScope.$broadcast('reloadFavorites');
  if (isFunction(callback)) { callback() };
});
```

Reply:
@arikfr Fixed.